zipline/finance/commission.py

```diff
@@ -69,6 +69,18 @@ def calculate(self, order, transaction):
         """
         raise NotImplementedError('calculate')
 
+    def __eq__(self, other):
+        return self.asdict() == other.asdict()
```

Comment:
A few notes here:

1. I think this `__eq__` implementation will currently incorrectly compare equal for objects of different types but with the same instance dictionary.  That's actually a realistic case too, b/c we have futures/equities versions of some models that are otherwise very similar.  The easy fix for this is to check if `type(other) == type(self)` along with whatever other comparisons are being made.  However...
2. In general, I don't think it's safe to just use comparison against `__dict__` for equality comparisons.  In particular, this implementation will only work if commission models only ever hold objects with nicely-behaved `__eq__` implementations/  Off the top of my head I can think of at least three cases of objects that can't be stored on a commission model without breaking this implementation: NaN, numpy/pandas objects, and custom objects that don't define `__eq__`.

The NaN case means that equality will almost always return false if one of the dictionary values is `NaN`:
```
In [23]: data = np.full(5, np.nan)

In [24]: d1 = {'a': 1, 'b': data[0]}

In [25]: d2 = {'a': 1, 'b': data[1]}

In [26]: d1 == d2
Out[26]: False
```

I say **almost**, because the dictionary constructor will short-circuit on object identity, so you can actually sometimes get the expected result depending on where your NaNs came from:

```
In [27]: d1 = {'a': 1, 'b': np.nan}

In [28]: d2 = {'a': 1, 'b': np.nan}

In [29]: d1 == d2
Out[29]: True
```

w/r/t numpy/pandas objects, `__eq__` returns a new numpy/pandas object, which will then raise an error when the dict comparator tries to call `__bool__`.

w/r/t other objects, there's no guarantee in general that every instance attribute of every commission model implements comparison the way you'd expect.  Notably, for example, the [`DummyMapping` class used by `PerContract`](https://github.com/quantopian/zipline/pull/1777/files#diff-0c146aa0be7e8ccc3c741f35e4928fb4R178) doesn't implement a `__eq__`, so this implementation will probably say that all `PerContract` instances are unequal.  (I think we also have a similar bug with the `VolatilityVolumeShare` slippage model)